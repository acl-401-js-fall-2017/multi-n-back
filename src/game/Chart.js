import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import RC2 from 'react-chartjs2';

export class Chart extends PureComponent {

  render() {
    const finalStats = this.props.game.finalStats;
    const nBacks = finalStats.game.sequences.map((sequence) => sequence.nBack).reverse();
    nBacks.pop();
    const chartData = {
      labels: new Array(finalStats.game.sequences.length+1).fill(1).map((e, i) => i+1),
      datasets: [{
        label: 'N',
        fill: false,
        data: nBacks,
        // backgroundColor: '#770D00',
        borderColor: ' #770D00',
        borderWidth: 2,
        pointRadius: 3,
        scaleStartValue: 0
      }]
      // options: {
      //   scales: {
      //     yAxes: [{
      //       ticks: {
      //         max: 10,
      //         min: 0
      //       }
      //     }]
      //   },
      //   legend: {
      //     display: true,
      //     labels: {
      //       boxWidth: 10,
      //       padding: 5
      //     }
      //   },
      //   title: 'No. of Sequence Played'
      // }
    };

    return (finalStats.game.sequences ?
      (
        <div className="game-chart">
          <RC2 data={chartData} type="line" width="500" height="300" redraw="true" />
          {/* <RC2 data={finalStats.game.sequences.nBack} type="line" width="500" height="300" redraw /> */}
        </div>
      ) : (
        <div className="game-chart"></div>
      )
    );
  }
}

export default connect(
  state => ({
    game: state.game
  }),
  null
)(Chart);