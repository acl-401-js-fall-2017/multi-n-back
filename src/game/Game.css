.game {
    margin: 1em auto;
    height: 20em;
    width: 20em;
}