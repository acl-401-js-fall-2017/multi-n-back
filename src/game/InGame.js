import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

class InGame extends PureComponent {
  render() {
    return (
      <div className="in-game">
        <h3>InGame component</h3>
      </div>
    );
  }
}

export default connect(
  state => ({}),
  null
)(InGame);