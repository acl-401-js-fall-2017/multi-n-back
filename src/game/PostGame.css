.post-game {
  height: 100%;
  width: 100%;
}