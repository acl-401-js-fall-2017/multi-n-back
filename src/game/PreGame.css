.pre-game {
    box-sizing: border-box;
    height: 100%;
    width: 100%;
    padding: 1em;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    color: #254044;
    font-size: 100%;
}

.pre-game h2 {
    margin: 0 auto;
    width: 8em;
    border-bottom: 2px solid #770d00;
    padding-bottom: 0.3em;

}

.pre-game section {
    display: flex;
    justify-content: space-between;
}

.pre-game section>fieldset {
    width: 38%;
}

.pre-game fieldset {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

.pre-game fieldset div {
    position: relative;
}

.pre-game input[type="radio"] {
    position: absolute;
    left: 20%;
    height: 3em;
    width: 3em;
    border-radius: 50%;
}

.pre-game label {
    display: inline-block;
    position: relative;
    left: 4%;
    height: 1.5em;
    width: 1.5em;
    letter-spacing: 0.1em;
}

.pre-game fieldset {
    border-radius: 0.4em;
}

/* TODO: add alt radio buttons; */

.pre-game .select-fieldset {
    position: relative;
}

.pre-game select {
    position: absolute;
    top: 40%; left: 50%;
    transform: translateX(-50%);
    font-size: 150%;
    background: #d8eaec;
    border-radius: 0.3em;
    outline: none;
    border: none;
    padding: 0.1em 0.4em;
}


.pre-game .play-button {
    margin-top: 0.5em;
    border-radius: 0.4em;
    border: none;
    padding: 0.5em 2em 0.5em 3em;
    background: #568289;
    color: #d8eaec;
    letter-spacing: 1em;
    font-family: 'Gruppo', cursive;
    font-size: 100%;
    font-weight: bold;
}

.pre-game .play-button:focus {
    background-image: -webkit-gradient(radial, center, from(#d8eaec), to(end-color));
    background-image: -webkit-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
    background-image: -moz-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
    background-image: radial-gradient(at center, #ffffff00 60%, #d8eaec);
}