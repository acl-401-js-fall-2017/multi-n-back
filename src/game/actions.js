import * as actions from '../services/constants';

this.game = {
  difficulty: {
    initN: 2,
    interval: 1000
  },
  numVariates: 4,
  status: 'in',
  sequences: [{
    variates: {
      position: true,
      number: true,
      audio: false,
      shape: true,
      color: true
    },
    nBack: 3,
    interval: 1000,
    combos: [{
      position: 6,
      number: 3,
      shape: 'square',
      color: 'red'
    }],
    comboPointer: 0
  }]
};


export function loadGame() {
  return {
    type: actions.INIT_GAME
  };
}

// export function gameSettings(game) {
//   return {
//     type: actions.GET_SETTINGS,
//     payload: 
//   };
// }