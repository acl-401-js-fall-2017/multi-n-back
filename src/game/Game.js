import React, { PureComponent } from 'react';
import { Switch, Route } from 'react-router';
import { connect } from 'react-redux';

import PreGame from './PreGame';
import InGame from './InGame';
import PostGame from './PostGame';

class Game extends PureComponent {
  
  // set up fake data
  constructor() {
    super();
    this.state = {
      difficulty: {
        initN: 2,
        interval: 1000
      },
      numVariates: 4,
      status: 'post',
      sequences: [{
        variates: {
          position: true,
          number: true,
          audio: false,
          shape: true,
          color: true
        },
        nBack: 3,
        interval: 1000,
        combos: [{
          position: 6,
          number: 3,
          shape: 'square',
          color: 'red'
        }],
        comboPointer: 0
      }],
      finalStats: []
    };
  }

  render() {
    console.log('in Game', this.state)
    const { match: { url } } = this.props;
    return (
      <main className="game">
        <h2>Game component</h2>
        {this.state.status === 'pre' &&
          <PreGame tempGame={this.state}/>
        }
        {this.state.status === 'in' &&
          <InGame tempGame={this.state}/>
        }
        {this.state.status === 'post' &&
          <PostGame tempGame={this.state}/>
        }
      </main>
    );
  }
}

export default connect(
  state => ({}),
  null
)(Game);