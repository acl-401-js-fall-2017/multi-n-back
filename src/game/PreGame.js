import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { setSettings } from './actions/actions';

import './PreGame';


class PreGame extends PureComponent {

  submitSettingsHandler = event => {
    event.preventDefault();
    const difficulty = event.target.difficulty.value;
    const numVariates = event.target.numVariates.value;
    this.props.setSettings(difficulty, numVariates);
    
  }

  render() {
    return (
      <div className="pre-game">
        <h3>PreGame component</h3>
        <form onSubmit={this.submitSettingsHandler}>
          <fieldset>
            <legend>Variate(using this name for now) Selection</legend>
            <select name="numVariates">
              <option value="2">2</option>
              <option value="3">3</option>
              <option value="4">4</option>
              <option value="5">5</option>
            </select>
          </fieldset>
          <fieldset>
            <legend>Difficulty Selection</legend>
            <div>
              <input type="radio" name="difficulty" id="difficulty_easy" value="easy" defaultChecked/>
              <label htmlFor="difficulty_easy">Easy</label>
            </div>
            <div>
              <input type="radio" name="difficulty" id="difficulty_medium" value="medium"/>
              <label htmlFor="difficulty_medium">Medium</label>
            </div>
            <div>
              <input type="radio" name="difficulty" id="difficulty_hard" value="hard"/>
              <label htmlFor="difficulty_hard">Hard</label>
            </div>
          </fieldset>
          <input type="submit"/>
        </form>
      </div>
    );
  }
}

export default connect(
  state => ({}),
  { setSettings }
)(PreGame);