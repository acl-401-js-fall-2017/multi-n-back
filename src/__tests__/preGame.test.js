import React from 'react';
import { shallow } from 'enzyme';
import { PreGame } from '../game/PreGame';

describe('PreGame', () => {
  it('shows PreGame component', () => {
    const wrapper = shallow(
      <PreGame/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});