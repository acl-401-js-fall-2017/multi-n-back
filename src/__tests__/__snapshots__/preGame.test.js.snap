// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreGame shows PreGame component 1`] = `
<form
  className="pre-game"
  onSubmit={[Function]}
>
  <h2>
    Settings
  </h2>
  <fieldset>
    <legend>
      Difficulty Selection
    </legend>
    <div>
      <input
        defaultChecked={true}
        id="difficulty_easy"
        name="difficulty"
        type="radio"
        value="easy"
      />
      <label
        htmlFor="difficulty_easy"
      >
        Easy
      </label>
    </div>
    <div>
      <input
        id="difficulty_medium"
        name="difficulty"
        type="radio"
        value="medium"
      />
      <label
        htmlFor="difficulty_medium"
      >
        Medium
      </label>
    </div>
    <div>
      <input
        id="difficulty_hard"
        name="difficulty"
        type="radio"
        value="hard"
      />
      <label
        htmlFor="difficulty_hard"
      >
        Hard
      </label>
    </div>
  </fieldset>
  <section>
    <fieldset>
      <legend>
        Audio on/off
      </legend>
      <div>
        <input
          defaultChecked={true}
          id="audio_choice_off"
          name="audio_choice"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <label
          htmlFor="audio_choice_off"
        >
          Off
        </label>
      </div>
      <div>
        <input
          id="audio_choice_on"
          name="audio_choice"
          onChange={[Function]}
          type="radio"
          value="true"
        />
        <label
          htmlFor="audio_choice_on"
        >
          On
        </label>
      </div>
    </fieldset>
    <fieldset
      className="select-fieldset"
    >
      <legend>
        Variate Selection
      </legend>
      <select
        name="numVariates"
      >
        <option
          key="0"
          value={2}
        >
          2
        </option>
        <option
          key="1"
          value={3}
        >
          3
        </option>
        <option
          key="2"
          value={4}
        >
          4
        </option>
      </select>
    </fieldset>
  </section>
  <input
    className="play-button"
    type="submit"
    value="PLAY"
  />
</form>
`;
